bitcoin/exchange/bitcoinprices.go

```diff
@@ -89,11 +96,12 @@ func (b *BitcoinPriceFetcher) fetchCurrentRates() error {
 }
 
 type BitcoinAverage struct {
-	cache map[string]float64
+	cache  map[string]float64
+	client *http.Client
 }
 
 func (b *BitcoinAverage) fetch() (err error) {
```

Comment:
perhaps this client should be passed to the fetch() method instead of adding it to that struct.